frontend/src/utils/auth.test.js

```diff
@@ -0,0 +1,123 @@
+import axios from "axios";
+import Cookie from "js-cookie";
+import jwt_decode from "jwt-decode";
+import { act } from "react-dom/test-utils";
+import { describe, expect, it, vi } from "vitest";
+
+import { useAuthStore } from "../store/auth";
+import {
+  userLogin,
+  userRegister,
+  logout,
+  setUser,
+  isAccessTokenExpired,
+  setAuthUser,
+  getRefreshedToken,
+} from "./auth";
+
+describe("#userLogin", () => {
+  it(
+    "should log in a user with valid email and password",
+    act(async () => {
+      axios.post = vi.fn().mockResolvedValue({
+        data: { access: "access_token", refresh: "refresh_token" },
+        status: 200,
+      });
+
+      const result = await userLogin("fake_user@example.com", "fake_password");
+
+      // expect(setAuthUser).toHaveBeenCalledWith("access_token", "refresh_token");
+
+      expect(result).toEqual({
+        // response: { access: "access_token", refresh: "refresh_token" },
+        response: null,
```

Comment:
Why this is returning `Null`?

Reply:
Need to mock some of the `Auth` modules and should be good!